### ISSUE TITLE
Feature Request: Move Group to New Window

### ISSUE THREAD

**Issue by Red-Eagle-LXIX** [tabkit/tabkit2#140]
I'd really like a feature to do this natively.
Right click a group (as grouped by Tab Kit) and move the whole group to a new window.
At the moment shift dragging a group to a new window does not appear to work only moving the first tab (and sometimes losing the rest of the group, sometimes leaving it in the source window) [shift drag within the same window works fine].

My workaround has been to use Session Manager to save the current window
Open a tab in a new window
Start Session Manager in the new window
Load saved session option and select session
Unselect all tabs not related to the group
Append To Current Window

A very time consuming workaround when you've got a group of 20-30 tabs you want to save out of a few hundred.

Ideally Shift Dragging to new window would work, but an option such as this would be even better as one would not to create the new window one's self.  It should work the same as the default 'Move to New Window' but for the whole group not just a single tab.

**Comment by industriflow** [comment]
I agree this feature would be very useful. Shift-dragging and Ctrl-dragging minimized groups has never worked reliably for me for several years, using all recommended TK2 settings and Classic Theme Restorer, etc.

**Comment by GWHAYWOOD** [comment]
+1

The issue raised by the OP that sometimes all the tabs in the group which were NOT dragged to the new window has bitten me many times.  Does it have its own issue #?  If not I will open one.  It is very tedious to recover all the closed tabs.

**Comment by PikachuEXE** [comment]
I can't promise when I will start working on it though
This year probably? 😬

**Comment by GWHAYWOOD** [comment]
:)

**Comment by GWHAYWOOD** [comment]
This just bit me today, again.  As a workaround for not being able to move a group of tabs to a new window natively, I occasionally

(1) move a group of tabs from a window on desktop AB1 to a newly-created window on e.g. desktop AB1;

(2) move the newly-populated window to e.g. desktop AB2 which has an existing browser window on it, one which already contains some tabs;

(3) move all the tabs from the newly-created window to the existing window on desktop AB2.

Today this resulted in about half the tabs in the original tab group vanishing without trace before I had the chance to move them to the newly-created window. :(

Just to let you know that this one still bites. :/

All the same it's still great to have this tool (most of the time, thank you. :)

**Comment by PikachuEXE** [comment]
Let me see if I can get this started this year, coz I now mostly work on https://github.com/FreeTubeApp/FreeTube which I use more often

**Comment by GWHAYWOOD** [comment]
:)

**Comment by PikachuEXE** [comment]
More than 50% done, copying/moving tabs (in group) to specified window is working
Except no idea how to maintain the group tree level/structure after that yet (multi-window handling is rare in existing code...)

**Comment by GWHAYWOOD** [comment]
Sorry, my numbering is off in the previous message.  There were five issues but one of them got lost in editing and I can't now remember it.  It was something relatively minor but might have been helpful in debugging.  If I remember it I'll post again.

**Comment by GWHAYWOOD** [comment]
I just noticed that now when I group several tabs from somewhere inside an existing group, the new group no longer splits the original group into two.  Instead, the new group is moved to a collapsed tab placed just above the original group.  I like this behaviour, please try to keep it. :)

**Comment by PikachuEXE** [comment]
I suggest testing on a separate profile first to ensure it's not anything caused by existing data/tabs (whatever it means)
You can copy some URLs from a few groups and open them in another profile (using a text file as URL buffer?)

I don't have long running multi-window session to test~ (I don't use Palemoon this way)

> I like this behaviour, please try to keep it.

I totally not sure what I did but probably won't change it

**Comment by GWHAYWOOD** [comment]
Ah, the new group isn't always collapsed after moving, but that's OK.

Incidentally there are capital letters on the TabKit menu which I presume are keyboard shortcuts although I've never managed to get any of them to work.  "Group tabs from hete to" and "Collapse group" are both given the letter 'T'.

**Comment by GWHAYWOOD** [comment]
I wish I could edit a post.  That should read "Group tabs from here to".

**Comment by GWHAYWOOD** [comment]
"...You can copy some URLs from a few groups and open them in another profile (using a text file as URL buffer?)..."

I don't understand what you mean.  I don't know what a profile is and I don't know how to use a text file as a URL buffer.  I can create a text file which contains text taken from a browser window (which I have highlighted with the mouse and then hit CTRL-C to save it in an X clipboard which I can then insert into an edit buffer) if that's anything like what you mean.

**Comment by PikachuEXE** [comment]
If you are using windows you can download portable without caring about profile
If other OS then I gotta see if there is a guide for profile...

**Comment by GWHAYWOOD** [comment]
Apart from a few VMs (which have never had browsers installed, and come with multiple issues of their own) we're Linux only here.  Mostly Debian Bullseye, gradually transitioning to Bookworm.  Bullseye on my desktop.

**Comment by PikachuEXE** [comment]
You can visit `about:profiles` (paste in URL bar) to manage profiles

![image](https://github.com/user-attachments/assets/16d77dd8-1350-4c84-9ac6-19f0de539463)

I forgot how to start Palemoon with profile picker though

![image](https://github.com/user-attachments/assets/f1780833-8f75-46f0-bd94-618bf5bdd91e)

**Comment by GWHAYWOOD** [comment]
There seems to be a problem with |"Group tabs from here to..."

Sometimes when I use that facility all the tabs that I'm trying to group vanish, except for the current tab. :(

**Comment by PikachuEXE** [comment]
Separate issue
Also please see if you can reproduce with a new profile, hard to fix something that can't be reproduced
